draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
 string are implicitly tolerated by the Verifier, as the digest is computed over a
 predefined byte string and not over a JSON object.
 
-Since the encoding is based on JSON, all value types that are allowed in JSON
-are also allowed in the `v` property in the source string. This includes
-numbers, strings, booleans, arrays, and objects.
-
-It is important to note that the HS-Disclosures object containing the
-source string is neither intended nor suitable for direct consumption by
+It is important to note that the Disclosures are neither intended nor
+suitable for direct consumption by
 an application that needs to access the disclosed claim values. The
-HS-Disclosures object is only intended to be used by a Verifier to check
+Disclosures are only intended to be used by a Verifier to check
 the digests over the source strings and to extract the original JSON
 data. The original JSON data is then used by the application. See
-(#processing_model) for details.
+(#verifier_verification) for details.
 
 ## Format of an SD-JWT
 
 An SD-JWT is a JWT that MUST be signed using the Issuer's private key. The
-payload of an SD-JWT MUST contain the `sd_digests` and `sd_digest_derivation_alg` claims
-described in the following, and MAY contain a Holder's public key or a reference
+payload of an SD-JWT MUST contain the `sd_digest_derivation_alg` claim
+described in the following, MAY contain one or more selectively disclosable claims, and MAY contain a Holder's public key or a reference
 thereto, as well as further claims such as `iss`, `iat`, etc. as defined or
 required by the application using SD-JWTs.
 
-### `sd_digests` Claim (Digests of Selectively Disclosable Claims) {#sd_digests_claim}
-
-The property `sd_digests` MUST be used by the Issuer to include digests of the salted claim values for any claim that is intended to be selectively disclosable.
+### Selectively Disclosable Claims
 
-The Issuer MUST choose a new, cryptographically random salt value for
-each claim value. The salt value MUST then be encoded as a string. It is
-RECOMMENDED to base64url-encode the salt value.
+For each claim that is to be selectively disclosed, the Issuer creates a Disclosure object, hashes it, and includes the hash instead of the original claim in the SD-JWT, as described next. The Disclosures are then sent to the Holder.
 
-The Issuer MUST generate the digests over a JSON literal according to
-[@!RFC8259] that is formed by
-JSON-encoding an object with the following contents:
+#### Creating Disclosures {#creating_disclosures}
+The Issuer MUST create a Disclosure object for each selectively disclosable claim as follows:
 
- * REQUIRED with the key `s`: the salt value,
- * REQUIRED with the key `v`: the claim value (either a string or a more complex object, e.g., for the [@OIDC] `address` claim),
- * OPTIONAL, with the key `n`: the claim name (if Claim Name Blinding is to be used for this claim).
+ * Create an array of three elements in this order:
```

Comment:
I would emphasize the order in other places too...

just curious, any logic for the order?

Reply:
At which other places? 

I don't have a strong opinion on the order, but my logic is that the value should come last, as it can be a large object. So having the salt (constant length) and claim name (roughly same length all the time) first makes it easier to read the disclosures, should that be necessary.